File: Hacker Cup 2020/Qualification Round/D1 - Running on Fumes - Chapter 1/solution.py
"""
Solution of [D: Running on Fumes - Chapter 1]

Written by hepheir@gmail.com
2020-07-27 / Python 3.8.0
"""

# ================================================================
# Override stdin / stdout for validations.

import sys

filePaths = {
    'sample' : '2020/Qualification Round/D1 - Running on Fumes - Chapter 1/data/sample_input.txt',
    'valid' : '',
    'submission' : ''
}

inputFilePath = filePaths['sample']
outputFilePath = inputFilePath.replace('input', 'output')

sys.stdin = open(inputFilePath, 'r')
sys.stdout = open(outputFilePath, 'w')

# ================================================================
# Solution

nextCity = []
gasCost = []

# the number of jobs
T = int(input())

for jobNum in range(T):
    N, M = map(int, input().split())

    for n in range(N):
        Ci = int(input())
        nextCity.append(n+1)
        gasCost.append(Ci)
    
    print(f'Case #{jobNum+1}: {longestCombinedTimberInterval}')

File: Hacker Cup 2020/Qualification Round/D1 - Running on Fumes - Chapter 1/solution.c
#include <stdio.h>
#include <stdlib.h>

char* inputFilePath = "/Users/hepheir/_git/Facebook-Hacker-Cup/2020/Qualification Round/D1 - Running on Fumes - Chapter 1/data/sample_input.txt";
char* outputFilePath = "/Users/hepheir/_git/Facebook-Hacker-Cup/2020/Qualification Round/D1 - Running on Fumes - Chapter 1/data/sample_output.txt";

struct Node {
    struct Node* next;
    unsigned int gasCost;
};
typedef struct Node City;


int main() {
    unsigned int T;
    unsigned int N, M;
    char buffer[32];

    City cities[1000000];
    FILE *fi, *fo;

    int i;

    
    fi = fopen(inputFilePath, "r");
    fo = fopen(outputFilePath, "w");

    // Get input
    fgets(buffer, 32, fi); sscanf(buffer, "%d", &T);
    fgets(buffer, 32, fi); sscanf(buffer, "%d %d", &N, &M);
    for (i = 0; i < N; i++) {
        fgets(buffer, 32, fi); sscanf(buffer, "%d", &(cities[i].gasCost));
    }

    fclose(fi);
    fclose(fo);
}